--- a/package/utils/bash/Makefile
+++ b/package/utils/bash/Makefile
@@ -30,7 +30,7 @@
   CATEGORY:=Utilities
   SUBMENU:=Shells
   TITLE:=The GNU Bourne Again SHell
-  DEPENDS:=+libncurses
+  DEPENDS:=+libncurses +libreadline
   URL:=http://www.gnu.org/software/bash/
 endef
 
@@ -44,6 +44,7 @@
 define Build/Configure
 	$(call Build/Configure/Default, \
 		--without-bash-malloc \
+		--with-installed-readline \
                 --bindir=/bin \
 	)
 endef
